csrc/dynamic_transform.cpp

```diff
@@ -492,7 +542,21 @@ void DynamicTransformConcretizer::mutate(TensorDomain* td) {
 
   Val* mutated_val = IrBuilder::create<TensorDomain>(
       td->container(), root_dom, rfactor_dom, domain, contig);
-  registerMutation(td, mutated_val);
+  registerConcretization(td, mutated_val);
+}
+
+void DynamicTransformConcretizer::mutate(IterDomain* id) {
+  auto new_val = maybeMutated(id);
+  TORCH_CHECK(
+      new_val->isA<IterDomain>(),
+      "Concretized IterDomain must not be an IterDomain. Found ",
+      new_val->toString());
+  auto new_id = new_val->as<IterDomain>();
+  TORCH_CHECK(
+      !new_id->isSymbolic(),
+      "Concretized IterDomain must not be Symbolic. Found ",
```

Comment:
Note that this check is not strictly necessary, but it is probably good housekeeping to ensure that the concretized `Fusion` does not include any `Symbolic` IDs. Note that it is currently failing in `DynamicTransform10_CUDA`. In that case the output `TensorView` even contains a symbolic `IterDomain` in its root domain after concretization:
```
Before Concretization:
Inputs:
  T0_g[ iS0{i0}, iS1{i2} ], float
Outputs:
  T2_g[ ?S6{i3}, ?S8{( ( i3 - 1 ) - 1 )}rf ], float
%kernel_math {
T1_l[ ?S4{i3}rf, ?S5{i4}rf ] = view( T0_g[ iS0{i0}, iS1{i2} ] )
i11 = i3 - 1;
T2_g[ ?S6{i3}, ?S8{( ( i3 - 1 ) - 1 )}rf ]
   = slice( T1_l[ ?S4{i3}rf, ?S5{i4}rf ], { {0, i3, 1} {1, i11, 1} } )
}
DynamicTransformConcretizationInfo
  Reshape:
    T1_l[ ?S4{i3}rf, ?S5{i4}rf ], { broadcast: 0 0, squeeze: 0 0, transforms: Merge at index: 0, Split Index at: 0 by: 4 }
  Resize:
    ?S8{( ( i3 - 1 ) - 1 )}rf, i
After Concretization:
Inputs:
  T0_g[ iS0{i0}, iS1{i2} ], float
Outputs:
  T2_g[ iS17{i3}, iS19{( ( i3 - 1 ) - 1 )}rf ], float
%kernel_math {
T3_l[ iS14{4}rf, iS15{( ceilDiv(( i0 * i2 ), 4) )}rf ] = view( T0_g[ iS0{i0}, iS1{i2} ] )
i11 = i3 - 1;
T2_g[ iS17{i3}, iS19{( ( i3 - 1 ) - 1 )}rf ]
   = slice( T3_l[ iS14{4}rf, iS15{( ceilDiv(( i0 * i2 ), 4) )}rf ], { {0, i3, 1} {1, i11, 1} } )
}
```

It is a little difficult to check for symbolic replacements at every stage of mutation, so it might be better left as a separate check at the end of concretization.